OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -0,0 +1,256 @@
+                            dataCallback)
+                } else {
+                    list.removeAt(i)
+                }
+            }
+            //Delete all videos which are marked as Deactivated in the database
+            environment?.storage?.deleteAllUnenrolledVideos()
+        }
+    }
+
+    private val dataCallback: DataCallback<Int> = object : DataCallback<Int>() {
+        override fun onResult(result: Int) {}
+        override fun onFail(ex: Exception) {
+            logger.error(ex)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (refreshOnResume) {
+            loadData(showProgress = false, isCached = true)
+            refreshOnResume = false
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        EventBus.getDefault().unregister(this)
+    }
+
+    fun onEventMainThread(event: EnrolledInCourseEvent?) {
+        refreshOnResume = true
+    }
+
+    private fun loadData(showProgress: Boolean, isCached: Boolean) {
+        if (showProgress) {
+            binding.loadingIndicator.root.visibility = View.VISIBLE
+            errorNotification.hideError()
+        }
+        if (isCached) {
+            getUserEnrolledCourses(api?.enrolledCoursesFromCache!!)
+        }
+        getUserEnrolledCourses(api?.enrolledCourses!!)
+    }
+
+    private fun invalidateView() {
+        binding.swipeContainer.isRefreshing = false
+        binding.loadingIndicator.root.visibility = View.GONE
+
+        if (!EventBus.getDefault().isRegistered(this@MyCoursesListFragment)) {
+            EventBus.getDefault().registerSticky(this@MyCoursesListFragment)
+        }
+    }
+
+    private fun showError(error: Throwable) {
+        context?.let { context ->
+            error.cause?.let { cause ->
+                errorNotification.showError(context, cause, R.string.lbl_reload) {
+                    if (NetworkUtil.isConnected(context)) {
+                        onRefresh()
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getUserEnrolledCourses(enrolledCoursesApi: Call<List<EnrolledCoursesResponse>>) {
+        enrolledCoursesApi.enqueue(object : Callback<List<EnrolledCoursesResponse>> {
+            override fun onResponse(call: Call<List<EnrolledCoursesResponse>>, response: Response<List<EnrolledCoursesResponse>>) {
+                if (response.isSuccessful && response.code() == HttpStatus.OK) {
+                    populateCourseData(ArrayList(response.body()))
+                } else if (response.code() == HttpStatus.UNAUTHORIZED) {
+                    environment?.router?.forceLogout(context,
+                            environment?.analyticsRegistry,
+                            environment?.notificationDelegate)
+                } else {
+                    showError(HttpStatusException(Response.error<Any>(HttpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
there should be a `Ref` for this inspiration.

Reply:
pending

Reply:
inspiration from [BaseWebViewFragment.java L:173](https://github.com/edx/edx-app-android/blob/1297a91ad45d4595e3de687318a19a026953fd87/OpenEdXMobile/src/main/java/org/edx/mobile/view/BaseWebViewFragment.java#L173)